dlt/destinations/filesystem/filesystem.py

```diff
@@ -82,16 +83,22 @@ def __init__(self, schema: Schema, config: FilesystemDestinationClientConfigurat
         # verify files layout. we need {table_name} and only allow {schema_name} before it, otherwise tables
         # cannot be replaced and we cannot initialize folders consistently
         self.table_prefix_layout = path_utils.get_table_prefix_layout(config.layout)
-
-    @property
-    def dataset_path(self) -> str:
-        ds_path = posixpath.join(self.fs_path, self.config.normalize_dataset_name(self.schema))
-        return ds_path
+        self.dataset_path = posixpath.join(self.fs_path, self.config.normalize_dataset_name(self.schema))
 
     def drop_storage(self) -> None:
         if self.is_storage_initialized():
             self.fs_client.rm(self.dataset_path, recursive=True)
 
+    @contextmanager
+    def with_staging_dataset(self) -> Iterator["FilesystemClient"]:
```

Comment:
why we need this in FileSystemClient?

Reply:
because we need a location where the "normal" athena table file live before they are copied into iceberg. The folder in the "final destination" for this table will be fully managed by iceberg, and we do not manually manipulated the files there.